ethcore/src/ethereum/ethash.rs

```diff
@@ -509,5 +510,14 @@ mod tests {
 		}
 	}
 
+	#[test]
+	fn test_difficulty_to_boundary() {
+		assert_eq!(Ethash::difficulty_to_boundary(&U256::from(0)), H256::from(0));
+		assert_eq!(Ethash::difficulty_to_boundary(&U256::from(1)), H256::from(0));
```

Comment:
condition is wrong - right side should really be `2^256`, though we might better approximate that in 256 bits with `2^256 - 1`.